### PR TITLE
Maintenance updates

### PR DESCRIPTION
**Pull request by greg-hellings**
Updates to match us to new best practices in meta_skeleton
Updates to role to run slightly better
Updates to molecule to allow tests to completely execute

### CODE REVIEW

**Comment by greg-hellings** [comment]
Failed downstream testing due to needing a value for OS_CLOUD to be defined in the new provisioner role for molecule_openstack_ci

File: molecule/shared/prepare.yml [hunk 1]
```diff
@@ -14,6 +11,10 @@
           - rh-gluster-3-samba-for-rhel-7-server-rpms
           - rh-gluster-3-nfs-for-rhel-7-server-rpms
       rhsm_unregister: true
+      rhsm_username: "{{ lookup('env', 'OASIS_RHSM_USERNAME') }}"
+      rhsm_password: "{{ lookup('env', 'OASIS_RHSM_PASSWORD') }}"
+      rhsm_server_hostname: "{{ lookup('env', 'OASIS_RHSM_SERVER_HOSTNAME') }}"
+      rhsm_auto_reattach: true
```
Comment (seandst) [inline molecule/shared/prepare.yml]: `rhsm_auto_attach`

**Comment by greg-hellings** [comment]
This PR now passes upstream and downstream testing. Please review.

File: molecule/shared/prepare.yml (hunk 1 above)
Comment (greg-hellings) [reply molecule/shared/prepare.yml]: Thanks.

**Comment by greg-hellings** [comment]
One more ping to everyone to get some review action moving.

File: molecule/default/molecule.yml [hunk 2]
```diff
@@ -35,10 +31,36 @@ provisioner:
...
+    - dependency
+    - destroy
+  prepare_sequence:
+    - dependency
+    - prepare
+  test_sequence:
+    - lint
+    - dependency
+    - destroy
+    - dependency
```
Comment (seandst) [inline molecule/default/molecule.yml]: https://github.com/ansible/molecule/pull/1739 is merged, so it's probably worth making a ticket to remove all this once the new version of molecule comes out, whatever version that might be.

Comment (greg-hellings) [reply molecule/default/molecule.yml]: Aye aye, captain. https://github.com/oasis-roles/gluster_gdeploy/issues/9

**Pull request merged**